lib/backend/k8s/resources/profile.go

```diff
@@ -240,7 +248,15 @@ func (c *profileClient) Watch(ctx context.Context, list model.ListInterface, rev
 	if len(rlo.Name) != 0 {
 		log.WithField("name", rlo.Name).Debug("Watching a single profile")
 		var err error
-		if strings.HasPrefix(rlo.Name, conversion.NamespaceProfileNamePrefix) {
+
+		// If we're watching the default-allow profile by name, then we exit
+		// with a new profileWatcher now and use fake watches in place
+		// of real ns and sa watchers. This profile watcher will never get any
+		// events because no events will occur for the static default-allow profile.
+		if rlo.Name == resources.DefaultAllowProfileName {
+			log.WithField("rv", revision).Debug("Creating watch on default-allow profile")
```

Comment:
Might want to put the word "fake" in this log, so as not to confuse some unknowing victim in the future. :p

Reply:
Good call :) 98b6175235ed48e801f1aa53c5f4954ab049949b